deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -698,14 +698,25 @@ def clone_cluster(request, src_name, src_stage):
     except NotAuthorizedException as e:
         log.error("Have an NotAuthorizedException error {}".format(e))
         if external_id is not None:
-            environs_helper.delete_nimbus_identifier(request, external_id)
+            try:
+                environs_helper.delete_nimbus_identifier(request, external_id)
+            except Exception as detail:
+                message = 'Failed to delete Nimbus identifier {}. Please verify that identifier no longer exists, Error Message: {}'.format(external_id, detail)
+                log.error(message)
+                return HttpResponse(message, status=403, content_type="application/json")
```

Comment:
I wouldn't do this. The original exception `NotAuthorizedException` is not returned to users. Also, 403 is misleading for the message `Failed to delete Nimbus identifier`.

Reply:
I can remove this. I think it makes more sense to just log the message. The stage still get created but the cluster is not cloned.

Reply:
If the user intention is to clone a cluster, I don't think it's useful to create a stage without the cloned cluster.